tests/common/features/active_enum.rs

```diff
@@ -0,0 +1,46 @@
+use sea_orm::entity::prelude::*;
+
+#[derive(Clone, Debug, PartialEq, DeriveEntityModel)]
+#[sea_orm(table_name = "active_enum")]
+pub struct Model {
+    #[sea_orm(primary_key)]
+    pub id: i32,
+    pub category: Option<Category>,
+    pub color: Option<Color>,
+    pub tea: Option<Tea>,
+}
+
+#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
+pub enum Relation {}
+
+impl ActiveModelBehavior for ActiveModel {}
+
+#[derive(Debug, Clone, PartialEq, DeriveActiveEnum)]
+#[sea_orm(rs_type = "String", db_type = "String(Some(1))")]
+pub enum Category {
+    #[sea_orm(string_value = "B")]
+    Big,
+    #[sea_orm(string_value = "S")]
+    Small,
+}
+
+#[derive(Debug, Clone, PartialEq, DeriveActiveEnum)]
+#[sea_orm(rs_type = "i32", db_type = r#"Integer"#)]
+pub enum Color {
+    #[sea_orm(num_value = 0)]
+    Black,
+    #[sea_orm(num_value = 1)]
+    White,
+}
+
+#[derive(Debug, Clone, PartialEq, DeriveActiveEnum)]
+#[sea_orm(
+    rs_type = "String",
+    db_type = r#"Enum("tea".to_owned(), vec!["EverydayTea".to_owned(), "BreakfastTea".to_owned()])"#
```

Comment:
Oh, this shouldn't be the intended usage?

Reply:
This allow create_enum_from_entity and create_table_from_entity got the required info and do their job. I'm thinking how to pull the enum variants (Vec<String>) part out.

Reply:
Yup it certainly looks clumsy

Reply:
Agree

Reply:
To avoid duplication of information, we should be able to delegate the value definitions to the actual enum definition.

Reply:
Probably storing `Box<Trait>`

Reply:
> Probably storing Box<Trait>

I think it's not possible...

```
error[E0038]: the trait `active_enum::ActiveEnumIden` cannot be made into an object
   --> src/entity/column.rs:68:17
    |
68  |     EnumTry(Box<dyn ActiveEnumIden>),
    |                 ^^^^^^^^^^^^^^^^^^ `active_enum::ActiveEnumIden` cannot be made into an object
    |
note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
   --> src/entity/active_enum.rs:112:27
    |
112 | pub trait ActiveEnumIden: Clone {
    |           --------------  ^^^^^ ...because it requires `Self: Sized`
    |           |
    |           this trait cannot be made into an object...
```

While we have to implement `Clone` for the trait, because...

```rust
#[derive(Debug, Clone, PartialEq)]
pub enum ColumnType { ... }
```

Reply:
If we require ActiveEnum to extend Iterable, then we can provide a default impl that returns all values.

```rust
fn values() -> Vec<String> {
    Self::IntoIter()...
}
```

Then the ColumnDef simply call this at runtime.

Reply:
Sounds like a good plan

Reply:
See bb78a1d